controllers/dashboard_controller.go

```diff
@@ -307,6 +307,9 @@ func (r *GrafanaDashboardReconciler) onDashboardDeleted(ctx context.Context, nam
 				var notFound *dashboards.GetDashboardByUIDNotFound
 				if !errors.As(err, &notFound) {
 					return err
+				} else {
```

Comment:
Nice catch! :) I would only suggest to slightly rewrite the code to make it easier to understand, e.g.:

```go
			resp, err := grafanaClient.Dashboards.GetDashboardByUID(*uid)
			if err != nil {
				var notFound *dashboards.GetDashboardByUIDNotFound
				if errors.As(err, &notFound) {
					return nil
				}

				return err
			}
```
WDYT?

Reply:
@weisdd updated now